rclpy/rclpy/impl/implementation_singleton.py

```diff
@@ -28,5 +28,17 @@
 
 import importlib
 
-
-rclpy_implementation = importlib.import_module('._rclpy', package='rclpy')
+module_name = '._rclpy'
+
+try:
+    rclpy_implementation = importlib.import_module(module_name, package='rclpy')
+except ImportError as e:
+    import os
```

Comment:
This import should at module level. No need to do it in this local scope. Same for `sys` below.